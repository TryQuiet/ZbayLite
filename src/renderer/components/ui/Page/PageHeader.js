import React from 'react'
import PropTypes from 'prop-types'

import Grid from '@material-ui/core/Grid'
import { withStyles } from '@material-ui/core/styles'

const styles = (theme) => ({
  root: {
    background: theme.palette.colors.white,
    order: -1,
    zIndex: 10,
    WebkitAppRegion: process.platform === 'win32' ? 'no-drag' : 'drag'
  }
})

export const PageHeader = ({ classes, children }) => (
  <Grid item className={classes.root}>
    {children}
  </Grid>
)

PageHeader.propTypes = {
  classes: PropTypes.object.isRequired,
  children: PropTypes.element.isRequired
}

export default React.memo(withStyles(styles)(PageHeader))
