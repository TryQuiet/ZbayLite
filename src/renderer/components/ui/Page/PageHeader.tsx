import React, { ReactElement } from 'react'

import Grid from '@material-ui/core/Grid'
import { makeStyles } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  root: {
    background: theme.palette.colors.white,
    order: -1,
    zIndex: 10,
    WebkitAppRegion: process.platform === 'win32' ? 'no-drag' : 'drag'
  }
}))

interface PageHeaderProps {
  children: ReactElement
}

export const PageHeader: React.FC<PageHeaderProps> = ({ children }) => {
  const classes = useStyles({})
  return (
    <Grid item className={classes.root}>
      {children}
    </Grid>
  )
}

export default PageHeader
