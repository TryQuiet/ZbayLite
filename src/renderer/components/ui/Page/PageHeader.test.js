/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { PageHeader } from './PageHeader'
import { mockClasses } from '../../../../shared/testing/mocks'
describe('PageHeader', () => {
  const Content = () => <div>Test Header</div>

  it('renders component', () => {
    const result = shallow(
      <PageHeader classes={mockClasses} >
        <Content />
      </PageHeader>
    )
    expect(result).toMatchSnapshot()
  })
})
