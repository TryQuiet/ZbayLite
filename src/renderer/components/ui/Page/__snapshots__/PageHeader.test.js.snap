// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders component 1`] = `
<WithStyles(ForwardRef(Grid))
  className="root-mock"
  item={true}
>
  <Content />
</WithStyles(ForwardRef(Grid))>
`;
